lib/puppet/util/windows/adsi.rb

```diff
@@ -162,6 +162,7 @@ def commit
       begin
         native_user.SetInfo unless native_user.nil?
       rescue Exception => e
+        raise Puppet::Error.new( "Puppet is not able to create or delete Active Directory users with the user resource", e ) if /8007089A in Active Directory/ =~ e.message
```

Comment:
Hard to know for sure. Catching it earlier could set us up for more validation of what the intent was. There are known scenarios you can work with a domain user in, just not ensure => present / absent.